llvm/lib/IR/AutoUpgrade.cpp

```diff
@@ -5387,6 +5387,14 @@ std::string llvm::UpgradeDataLayoutString(StringRef DL, StringRef TT) {
     return Res;
   }
 
+  // AArch64 data layout upgrades.
+  if (T.isAArch64()) {
+    // Add "-Fn32"
+    if (!DL.contains("-Fn32"))
+      Res.append("-Fn32");
```

Comment:
Ideally, the "fixed" datalayout string should be in the same order as the one a new compiler would generate (so if you're sticking it at the end here, stick it at the end on the actual strings).  I think we compare datalayouts using string equality in LTO.

Reply:
OK, thanks. I was afraid of that. I was copying two patterns - in ARM, `Fn8` seemed to be after any `p` and before `i`. In this file, the modifications are usually (always?) appending.

Reply:
Feel free to make the datalayout string whatever order is most convenient for the autoupgrade.